lib/src/set/set_builder.dart

```diff
@@ -63,6 +66,34 @@ class SetBuilder<E> {
     }
   }
 
+  /// Uses `base` as the collection type for all sets created by this builder.
+  ///
+  ///     // Iterates over elements in ascending order
+  ///     new SetBuilder<int>.using(() => new SplayTreeSet());
+  ///
+  ///     // Uses custom equality
+  ///     new SetBuilder<int>.using(() => new LinkedHashSet(
+  ///         equals: (a, b) => a % 255 == b % 255),
+  ///         hashCode: (n) => (n % 255).hashCode));
+  ///
+  /// The set returned by `base` must be empty, mutable, and each call must
+  /// instantiate and return a new object. The methods `difference`,
+  /// `intersection` and `union` of the returned set must create sets of the
+  /// same type.
+  ///
+  /// Use [withDefaultBase] to reset `base` to the default value.
+  void withBase(Set<E> Function() base) {
+    _setSafeSet(base()..addAll(_set));
+    _setConstructor = base;
+  }
+
+  /// As [withBase], but sets `base` back to the default value, which
+  /// instantiates `Set<E>`.
+  void withDefaultBase() {
+    _setConstructor = null;
+    _setSafeSet(_createSet()..addAll(_set));
```

Comment:
Any reason for updating _setConstructor first here, instead of last on line 86?

AFAICT both work--in which case please do the same in both methods :)

Reply:
No, `_createSet()` would still call the old `_setConstructor`.

The question should rather be: "Why doesn't `withBase()` look like this:"
```
void withBase(Set<E> Function() base) {
  _setConstructor = base;
  _setSafeSet(createBase()..addAll(_set)); // In my PR, I call `base` instead of `_createSet`
}
```
And the reason for that is: If a user calls `withBase` with `null`, it will throw an error *before* the builder is mutated.
... but an explicit null check is probably more readable

Reply:
That makes sense. Yes, an explicit null check would be clearer, I think.